plugins/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/condition/core/ResourceExists.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v 1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributor:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.reddeer.condition.core;
+
+import static org.junit.Assert.assertNotNull;
+
+import java.util.List;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Matcher;
+import org.jboss.reddeer.common.condition.AbstractWaitCondition;
+import org.jboss.tools.openshift.core.connection.Connection;
+
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.model.IResource;
+
+/**
+ * Wait condition to wait for existence of a 
+ * resource(s) of a given kind that match a given matcher.
+ * 
+ * @author adietish@redhat.com
+ *
+ */
+public class ResourceExists extends AbstractWaitCondition {
```

Comment:
I dont think that this makes a lot of sense. The little code there is in both and the little code that they can share vs the additionally required checks, switches etc.